### PR TITLE
Support container groups

### LINKED ISSUE #389: Container Instances - PQ3
Resource Support for Container Instances
Coverage:
1.Dev
2.Test
3.Documentation

### PR DESCRIPTION
**Pull request by sathish-progress**
### Description

support Azure container group (s)

### Issues Resolved
#389 

### Check List

- [X] New functionality includes tests
- [X] All tests pass
- [X] `rake lint` passes
- [X] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: test/integration/verify/controls/azure_container_group.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+resource_group = input(:resource_group, value: '')
+container_group_name = input(:inspec_container_group_name, value: '')
+
+control 'check container group properties' do
+  describe azure_container_group(resource_group: resource_group, name: container_group_name) do
```
Comment (soumyo13) [inline test/integration/verify/controls/azure_container_group.rb:5]: After the control statement. We should add the impact and the description. This is how we do in was. Do we have to do the same?

File: test/integration/verify/controls/azure_container_groups.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+control 'check container groups' do
+  describe azure_container_groups do
```
Comment (soumyo13) [inline test/integration/verify/controls/azure_container_groups.rb:2]: After the control block, we should add the impact and the description. This is how we do in was. Do we have to do the same?

File: test/integration/verify/controls/azure_container_groups.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    it { should exist }
+    its('types') { should include 'Microsoft.ContainerInstance/containerGroups' }
+    its('locations') { should include 'eastus' }
+    its('restart_policies') { should include 'OnFailure' }
+    its('os_types') { should include 'Linux' }
+  end
+end
+
+control 'check all container groups that have been provisioned' do
+  describe azure_container_groups.where(provisioning_state: 'Succeeded') do
```
Comment (soumyo13) [inline test/integration/verify/controls/azure_container_groups.rb:12]: After the control block, we should add the impact and the description. This is how we do in was. Do we have to do the same?

File: terraform/outputs.tf [hunk 4]
```diff
@@ -371,3 +371,8 @@ output "policy_definition_associated_cosmodb_id" {
 //  description = "the name of the policy exemption"
```
Comment (soumyo13) [inline terraform/outputs.tf:371]: If the commented lines are not required. Can we delete it?

File: terraform/azure.tf [hunk 5]
```diff
@@ -1232,7 +1232,38 @@ resource "azurerm_function_app" "web_app_function" {
...
+    name   = "setup-hw-tutorials"
+    image  = "mcr.microsoft.com/azuredocs/aci-tutorial-sidecar"
+    cpu    = "0.5"
+    memory = "0.5"
+  }
+
+  tags = {
+    environment = "inspec_trial"
+  }
+}resource "azurerm_policy_definition" "inspec_policy_definition" {
```
Comment (soumyo13) [inline terraform/azure.tf:1266]: Can we add a extra line after the ending } resource?

**Review by soumyo13 — APPROVED** [review]

**Comment by soumyo13** [comment]
Left out some comments. Please check once. LGTM.

File: terraform/outputs.tf (hunk 4 above)
Comment (sathish-progress) [reply terraform/outputs.tf:371]: no they are needed when the policy exemption tf is released and we can simply uncomment it

File: test/integration/verify/controls/azure_container_group.rb (hunk 1 above)
Comment (sathish-progress) [reply test/integration/verify/controls/azure_container_group.rb:5]: It is not mandatory .. and most integration tests in Azure either way do not follow that ..

File: test/integration/verify/controls/azure_container_groups.rb (hunk 2 above)
Comment (sathish-progress) [reply test/integration/verify/controls/azure_container_groups.rb:2]: It is not mandatory..

File: test/integration/verify/controls/azure_container_groups.rb (hunk 3 above)
Comment (sathish-progress) [reply test/integration/verify/controls/azure_container_groups.rb:12]: It is not mandatory..

**Review by IanMadd — APPROVED** [review]

**Pull request merged**